### DISCUSSION TITLE
Not seeing presentmentPrices?

### DISCUSSION THREAD

**Discussion by moreguppy** [gatsbyjs/gatsby#32090] [Ideas / Feature Requests] [answered]
I'm looking to get the `presentmentPrices` (which are the variant prices set to multiple countries/regions).

I can't see this being available in the source when using it in a gatsby-shopify-starter test project:

![image](https://user-images.githubusercontent.com/2033609/121910713-fb176d80-cd26-11eb-97bf-246666f17c41.png)

I don't think this is to do with shopify settings, because you can get the `presentmentPrices` via the shopify-buy sdk — so it appears as though the node is not resolving properly when being included as a gatsby plugin.

I can see this is included in `gatsby-source-shopify/src/queries.js` when looking at it in `node_modules`

```
const PRODUCTS_QUERY = `
  query GetProducts($first: Int!, $after: String) {
    products(first: $first, after: $after) {
      pageInfo {
        hasNextPage
      }
      edges {
        cursor
        node {
          # everything else...
          variants(first: 250) {
            edges {
              node {
                # everything else...
                presentmentPrices(first: 250) {
                  edges {
                    node {
                      price {
                        amount
                        currencyCode
                      }
                      compareAtPrice {
                        amount
                        currencyCode
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
`;
```

But I cannot see presentmentPrices in the source code: https://github.com/gatsbyjs/gatsby-source-shopify/blob/master/plugin/src/query-builders/products-query.ts

Thanks for reading!

**Comment by sslotsky** [comment]
It looks like what's in your `node_modules` is probably an older version of the plugin. The new version doesn't currently list the presentment prices because when we initially attempted it, we weren't sure how to handle connections that come back without IDs. But now I'm thinking we may be able to just attach them as a simple array.

**Comment by moreguppy** [comment]
Hi @sslotsky, I'm on `~4.7.1`, so that sounds about right.

> But now I'm thinking we may be able to just attach them as a simple array.

This would be great!

**Comment by kaushikdivyansh** [comment]
>  But now I'm thinking we may be able to just attach them as a simple array.

@sslotsky I know you all are super busy  but do we've a time frame for when `presentmentPrices` will be available again?

**Comment by sslotsky** [comment]
There is a temporary blocker here and that's query complexity. If I add `presentmentPrices` to the current products query then I'll go over the connection limit that's permitted by the bulk API. 

We can mostly likely solve this problem by breaking variants out into a separate operation, but we have to do that carefully to make sure that things like variant deletion still work on incremental builds, because that one is tricky.

The tl;dr is, we have to refactor some stuff, and we have a ticket for this refactoring that has not yet been prioritized. We appreciate your patience.

**Reply by moreguppy** [reply]
Hi @sslotsky any chance this will be in a future update soon? Is it even going to be possible with the bulk API limit?

Thanks for looking into this!

**Comment by jackonawalk** [comment, marked as answer]
Hi there, we're currently testing the next major version of gatsby-source-shopify. This may address your issue. Learn more about upgrading: https://www.notion.so/gatsbyjs/Instructions-for-upgrading-to-Shopify-7-e5cb1068b35b4de7ba7e9ffb160aad26
